src/nupic/algorithms/Connections.hpp

```diff
@@ -449,41 +441,46 @@ class Connections : public Serializable
    *
    * @retval Number of cells.
    */
-  CellIdx numCells() const;
+  size_t numCells() const { return cells_.size(); }
 
   /**
    * Gets the number of segments.
    *
    * @retval Number of segments.
    */
-  Segment numSegments() const;
+  size_t numSegments() const { 
+	  NTA_ASSERT(segments_.size() >= destroyedSegments_.size()); //TODO remove destroyedSegments_? 
+	  return segments_.size() - destroyedSegments_.size(); }
 
   /**
    * Gets the number of segments on a cell.
    *
    * @retval Number of segments.
    */
-  SegmentIdx numSegments(CellIdx cell) const;
+  size_t numSegments(CellIdx cell) const { return cells_[cell].segments.size(); }
 
   /**
    * Gets the number of synapses.
    *
    * @retval Number of synapses.
    */
-  Synapse numSynapses() const;
+  size_t numSynapses() const {
+    NTA_ASSERT(synapses_.size() >= destroyedSynapses_.size());
+    return synapses_.size() - destroyedSynapses_.size();
+  }
 
   /**
    * Gets the number of synapses on a segment.
    *
    * @retval Number of synapses.
    */
-  SynapseIdx numSynapses(Segment segment) const;
+  size_t numSynapses(Segment segment) const { return segments_[segment].synapses.size(); }
 
   /**
    * Comparison operator.
    */
-  bool operator==(const Connections &other) const;
-  bool operator!=(const Connections &other) const;
+  virtual bool operator==(const Connections &other) const;
+  inline bool operator!=(const Connections &other) const { return !operator==(other); }
```

Comment:
`virtual` is not needed, since this class should never be subclassed.  It's not supported, this is the only virtual method here. I expect if it were subclassed that a lot of things would break.

My understanding is that all modern compilers ignore `inline` and figuring out what to inline on its own.

Reply:
I'll remove the virtual, you're right about the inline, but it does not hurt either.

Reply:
`inline` is a nitpick, you can ignore me here.